tkupchyn/homework_03/ip_validation.py

```diff
@@ -0,0 +1,25 @@
+import re
+
+
+def is_valid_ip(ip_address: str) -> bool:
+
+    """
+    Checks if the given IP address is valid.
+
+    Args:
+        ip_address (str): The input string that may contain a mixture of digits and letters.
+
+    Returns:
+        bool: True if ip address is valid, False otherwise.
+    """
+
+    octets = ip_address.split('.')
+
+    if len(octets) != 4:
+        return False
+
+    for octet in octets:
+        if not octet.isdigit() or int(octet) not in range(0, 256) or re.match(r'^0\d', octet):
```

Comment:
can you avoid using regex?

Reply:
why you need   re.match(r'^0\d', octet) 
if you have 
 int(octet) in range(0, 256)         check

Reply:
Because string with IP address can be represented in different ways:
'123.045.067.089' - not valid
'123.45.67.89' - valid
and both of them match the condition int(octet) in range(0, 256) that's why I've used regex.